src/main/java/wolox/training/models/Book.java

```diff
@@ -106,14 +110,16 @@ public String getYear() {
     }
 
     public void setYear(String year) {
+        Preconditions.checkNotNull(year);
         this.year = year;
     }
 
     public int getPages() {
         return pages;
     }
 
-    public void setPages(int pages) {
+    public void setPages(Integer pages) {
+        Preconditions.checkNotNull(pages);
```

Comment:
How about some more conditions? For instance, you could establish that the number of pages must be greater than 10 🙂 
You can perform this validation using `Preconditions.checkArgument(...)`